test/Microsoft.SqlTools.ServiceLayer.Test.Common/TestConnectionProfileService.cs

```diff
@@ -0,0 +1,206 @@
+
+            if (connectionProfile != null)
+            {
+                //If the password is empty, get the credential using the service
+                if (string.IsNullOrEmpty(connectionProfile.Password))
+                {
+                    Credential credential = await ReadCredentialAsync(connectionProfile.formatCredentialId());
+                    connectionProfile.Password = credential.Password;
+                }
+
+                ConnectParams connenctParam = CreateConnectParams(connectionProfile, serverType, databaseName);
+               
+                return connenctParam;
+            }
+            return null;
+        }
+
+        /// <summary>
+        /// Request a Read Credential for given credential id
+        /// </summary>
+        private async Task<Credential> ReadCredentialAsync(string credentialId)
+        {
+            var credentialParams = new Credential();
+            credentialParams.CredentialId = credentialId;
+
+            ServiceTestDriver driver = Driver;
+            if (driver == null)
+            {
+                TestServiceProvider.InitializeTestServices();
+                return await Task.Factory.StartNew(() => CredentialService.Instance.ReadCredential(credentialParams));
+            }
+            else
+            {
+                return await Driver.SendRequest(ReadCredentialRequest.Type, credentialParams);
+            }
+        }
+
+        /// <summary>
+        /// Create a connection parameters object
+        /// </summary>
+        private ConnectParams CreateConnectParams(ConnectionProfile connectionProfile, TestServerType serverType, string databaseName)
+        {
+            ConnectParams connectParams = new ConnectParams();
+            connectParams.Connection = new ConnectionDetails();
+            connectParams.Connection.ServerName = connectionProfile.ServerName;
+            connectParams.Connection.DatabaseName = connectionProfile.Database;
+            connectParams.Connection.UserName = connectionProfile.User;
+            connectParams.Connection.Password = connectionProfile.Password;
+            connectParams.Connection.AuthenticationType = connectionProfile.AuthenticationType.ToString();
+            if (!string.IsNullOrEmpty(databaseName))
+            {
+                connectParams.Connection.DatabaseName = databaseName;
+            }
+            if (serverType == TestServerType.Azure)
+            {
+                connectParams.Connection.ConnectTimeout = 30;
+                connectParams.Connection.Encrypt = true;
+                connectParams.Connection.TrustServerCertificate = false;
+            }
+            return connectParams;
+        }
+
+        private static IEnumerable<TestServerIdentity> InitTestServerNames()
+        {
+            try
+            {
+                string testServerNamesFilePath = Environment.GetEnvironmentVariable("TestServerNamesFile");
```

Comment:
It'd be great to fall back on a default disk location (I think that's how most non-build machines will set this up)